libexec/rtld-elf/rtld.c

```diff
@@ -4709,48 +4709,89 @@ tls_get_addr_common(Elf_Addr **dtvp, int index, size_t offset)
 #if defined(__aarch64__) || defined(__arm__) || defined(__mips__) || \
     defined(__powerpc__) || defined(__riscv)
 
+/*
+ * Return pointer to allocated TLS block
+ */
+static void *
+get_tls_block_ptr(void *tcb, size_t tcbsize)
+{
+    size_t extra_size, post_size, pre_size, tls_block_size;
+    size_t tls_init_align;
+
+    tls_init_align = MAX(obj_main->tlsalign, 1);
+
+    /* Compute fragments sizes. */
+    extra_size = tcbsize - TLS_TCB_SIZE;
+#if defined(__aarch64__) || defined(__arm__)
+    post_size =  roundup2(TLS_TCB_SIZE, tls_init_align) - TLS_TCB_SIZE;
+#else
+    post_size = 0;
+#endif
+    tls_block_size = tcbsize + post_size;
+    pre_size = roundup2(tls_block_size, tls_init_align) - tls_block_size;
+
+    return ((char *)tcb - pre_size - extra_size);
+}
+
 /*
  * Allocate Static TLS using the Variant I method.
  */
```

Comment:
You did not add the comment here in the real rtld, and I would add the same "see also" reference to lib/libc/gen/tls.c.